### DISCUSSION TITLE
unable to get the correct context for modal bottom sheet

### DISCUSSION THREAD

**Discussion by shashkiranr** [rrousselGit/riverpod#2986] [Q&A]
Hi,

I am using the following code to override the provider using modal bottom sheet. I am unable to read the overridden value. I realised that context is wrong. 

```
showModalBottomSheet<void>(
      useSafeArea: true,
      context: context,
      isScrollControlled: true,
      builder: (BuildContext context) {
        return ProviderScope(
          overrides: [
            customBundleProvider
                .overrideWithValue(CustomBundleData(
              id: id,
              isAdded: isAdded,
            )),
          ],
          child: const CustomPage(), // Another Consumer Widget
        );
      },
    );
```

```
final customPlaylistBundleProvider = Provider<CustomBundleData>((ref) {
  return const CustomBundleData(id: 'id');
});
```

When I try to read the provider value I always get the default value of provider and not the overridden value.

How can I read the overridden value correctly?

Thank you.

**Comment by AhmedLSayed9** [comment]
You need to define the parent container as dialogs use different context:
```dart
builder: (BuildContext c) { /// This context named different from the other context
        return ProviderScope(
          parent: ProviderScope.containerOf(context),
          overrides: [
            customBundleProvider
                .overrideWithValue(CustomBundleData(
              id: id,
              isAdded: isAdded,
            )),
          ],
          child: const CustomPage(), // Another Consumer Widget
        );
      },
```
And you're overriding `customBundleProvider`  while the other provider is `customPlaylistBundleProvider`

**Reply by shashkiranr** [reply]
customPlaylistBundleProvider is wrong.  it is customBundleProvider so I am using the right Provider.  

I tried your suggestion of using `parent: ProviderScope.containerOf(parentContext),` but i am not able to access the overridden customBundleProvider value in CustomPage. 

I found a workaround by using StateProvider for now.
